server-common/src/main/java/org/apache/kafka/server/group/share/SharePartitionKey.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.group.share;
+
+import org.apache.kafka.common.Uuid;
+
+import java.util.Objects;
+
+/**
+ * Common immutable share partition key class. This class is
+ * placed in server-common so that it can be freely used across
+ * various modules.
+ */
+public class SharePartitionKey {
+    private final String groupId;
+    private final Uuid topicId;
+    private final int partition;
+
+    public SharePartitionKey(String groupId, Uuid topicId, int partition) {
+        this.groupId = groupId;
+        this.topicId = topicId;
+        this.partition = partition;
+    }
+
+    public static class Builder {
```

Comment:
Why do we need a builder for this class? Three arguments seems fine to deal with, and I doubt it will evolve over time. How about a since private constructor, and two public static factories: one (String, Uuid, int) and one (String, TopicIdPartition)

Reply:
Can't use `(String, TopicIdPartition)`. All coordinator keys use topic specified as Uuid.